libvirt: 6.3.0 -> 6.6.0

basvandijk opened the pull request:
This is https://github.com/NixOS/nixpkgs/pull/98937 targeted for staging.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

https://github.com/libvirt/libvirt/blob/master/NEWS.rst#v660-2020-08-02

###### Things done

Also upgrades `python3Packages.libvirt`
and patches the `rpcgen` tool from Apple's `developer_cmds` package.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

basvandijk committed:
libvirt: 6.3.0 -> 6.6.0

Also upgrades `python3Packages.libvirt`
and patches the `rpcgen` tool from Apple's `developer_cmds` package.

basvandijk commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -17,34 +17,38 @@ let
         fetchSubmodules = true;
       };
 
-  patches = [
-    ./0001-Fix-build-with-libtirpc.patch
```
Note to reviewers: this patch is in the 6.6.0 release.

basvandijk commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -17,34 +17,38 @@ let
+    darwin.developer_cmds # needed for rpcgen
   ];
 
-  nativeBuildInputs = [ makeWrapper pkgconfig docutils rpcsvc-proto ]
```
Note to reviewers: I made including `rpcsvc-proto` conditional on `!isDarwin` since it's not needed on that platform.

basvandijk commented on pkgs/development/libraries/libvirt/default.nix:
```diff
@@ -17,34 +17,38 @@ let
-    ++ optionals (!buildFromTarball) [ autoreconfHook ];
   buildInputs = [
     libxml2 gnutls perl python2 readline gettext libtasn1 libgcrypt yajl
-    libxslt xhtml1 perlPackages.XMLXPath curl libpcap glib dbus libtirpc
```
Note to reviewers: `libtirpc` is only needed on Linux so I moved it to the `optionals stdenv.isLinux` section below.

ajs124 commented:
The reason for going to 6.6.0 instead of 6.7.0 (which has been out for nearly a month) is that 6.7 uses meson, I assume?

TredwellGit reviewed (changes requested):
Please:
- Update `http://libvirt.org/` to `https://libvirt.org/` and `git://libvirt.org/libvirt.git` to `https://gitlab.com/libvirt/libvirt.git`.
- See if it works with `NIX_CFLAGS_COMPILE = "-fno-stack-protector";` removed.
- `"--with-runstatedir=/run" # TODO: remove when autoconf 1.70 is released` should say 2.70.
- See https://github.com/NixOS/nixpkgs/pull/85562

basvandijk committed:
libvirt: update homepage to https URL

basvandijk committed:
libvirt: update autoconf 1.70 comment to 2.70

basvandijk committed:
libvirt: remove -fno-stack-protector because it builds without it

basvandijk committed:
libvirt: update source URLs to https

basvandijk commented:
@TredwellGit 
> * Update http://libvirt.org/ to https://libvirt.org/ 

Done

> and git://libvirt.org/libvirt.git to https://gitlab.com/libvirt/libvirt.git.

The official repo [seems to be](https://libvirt.org/downloads.html#git) `https://libvirt.org/git/libvirt.git` so I changed it to that.

> * See if it works with NIX_CFLAGS_COMPILE = "-fno-stack-protector"; removed.

I removed it and can confirm it builds without it (I did test this on master and not on staging).

> * "--with-runstatedir=/run" # TODO: remove when autoconf 1.70 is released should say 2.70.

Done.

> * See #85562

Thanks for the reference.

basvandijk commented:
@ajs124 

> The reason for going to 6.6.0 instead of 6.7.0 (which has been out for nearly a month) is that 6.7 uses meson, I assume?

Indeed. I have a need for libvirt-6.6.0 because I have a server running that version and I didn't want to go to the trouble figuring out how a build with meson works.

I might look into packaging 6.7.0 after this.

TredwellGit commented:
>The official repo seems to be https://libvirt.org/git/libvirt.git so I changed it to that.

Go to https://libvirt.org/git/ and notice how it is a mirror of https://gitlab.com/libvirt/.

Please:
- Update `repositories.git`.
- Update `url` and `homepage` in the Python module.
- Update SysVirt in https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/perl-packages.nix.

basvandijk committed:
libvirt: fetch sources from gitlab

That seems to be the official repository now.

basvandijk commented:
@TredwellGit there [doesn't seem to be a libvirt-6.6.0 release](https://gitlab.com/libvirt/libvirt-perl/-/tags).

I just checked and `perlPackages.SysVirt` (6.3.0) does build correctly against `libvirt-6.6.0`. [Their commits](/nix/store/xr6566kc7yjgxlxhva7cqnh1kc8ls9yg-perl5.32.0-Sys-Virt-6.3.0) also indicate that the 6.3.0 perl bindings build against 6.6.0 and 6.7.0.  So I think the best course of action is to remove:

```
assert version == pkgs.libvirt.version; 
```

basvandijk committed:
perlPackages.SysVirt: remove assert version == pkgs.libvirt.version

SysVirt-6.3.0 should build correctly against libvirt-6.6.0 and also
libvirt-6.7.0 since their were no API changes.

Also note that because of this there's no SysVirt-6.6.0 release.

So this removes the `assert version == pkgs.libvirt.version` protection.

stigtsp commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -18419,8 +18419,8 @@ let
-    src = assert version == pkgs.libvirt.version; fetchgit {
-      url = "git://libvirt.org/libvirt-perl.git";
+    src = fetchgit {
+      url = "https://gitlab.com/libvirt/libvirt-perl.git";
```
Can this use `fetchFromGitLab`?

basvandijk replied:
TIL about `fetchFromGitLab`. I'll replace it.  Thanks.

stigtsp commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -18419,8 +18419,8 @@ let
     version = "6.3.0";
-    src = assert version == pkgs.libvirt.version; fetchgit {
-      url = "git://libvirt.org/libvirt-perl.git";
+    src = fetchgit {
+      url = "https://gitlab.com/libvirt/libvirt-perl.git";
       rev = "v${version}";
       sha256 = "0m75g0kf2dmllx5c1wxzszm1zakiqjighcsjbgq66674pvqf5mk7";
     };
```
Sorry, looking at it a bit closer, 6.3.0 is available on CPAN so it would probably be better to get the source from there.

```suggestion
   src = fetchurl {
      url = "mirror://cpan/authors/id/D/DA/DANBERR/Sys-Virt-v6.3.0.tar.gz";
      sha256 = "6333fe3c554322fec5a3e1890b08a4ea4f39b0fbb506b3592688a5785a488f39";
    };
```

basvandijk replied:
Agreed.

basvandijk committed:
perlPackages.SysVirt: fetch source from CPAN

Co-authored-by: Stig P <stig@stig.io>

basvandijk committed:
python3Packages.libvirt: fetch source using fetchFromGitLab

basvandijk commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -18419,10 +18419,9 @@ let
   SysVirt = buildPerlModule rec {
     pname = "Sys-Virt";
     version = "6.3.0";
-    src = assert version == pkgs.libvirt.version; fetchgit {
```
Note to reviewers: Sys-Virt-6.3.0 builds correctly with `libvirt-6.6.0` (and should also build with `libvirt-6.7.0`) because there were no API changes in those releases.  Because of this reason there are no Sys-Virt-6.6.0 or Sys-Virt-6.7.0 releases.
